luigi/execution_summary.py

```diff
@@ -30,36 +30,93 @@
+    =============================  =====  ==========================================================
+    
+    """
+    SUCCESS = 0
+    SUCCESS_WITH_RETRY = 1
+    FAILED = 2
+    FAILED_AND_SCHEDULING_FAILED = 3
+    SCHEDULING_FAILED = 4
+    NOT_RUN = 5
+    MISSING_EXT = 6
+
+    # Make sure all the retcodes have a (smiley, reason) in this dictionary
+    _DETAILS = {
+        SUCCESS: (":)", "there were no failed tasks or missing dependencies"),
+        SUCCESS_WITH_RETRY: (":)", "there were failed tasks but they all succeeded in a retry"),
+        FAILED: (":(" ,"there were failed tasks"),
+        FAILED_AND_SCHEDULING_FAILED: (":(", "there were failed tasks and tasks whose scheduling failed"),
+        SCHEDULING_FAILED: (":(", "there were tasks whose scheduling failed"),
+        NOT_RUN: (":|", "there were tasks that were not granted run permission by the scheduler"),
+        MISSING_EXT: (":|", "there were missing external dependencies")
+    }
 
 
 class LuigiRunResult:
     """
     Result of the execution (build/run) will be of type LuigiRunResult instead of
-    the regular Boolean response if [execution_summary]legacy_run_result is set
-    to False in the config.
+    the regular Boolean response if the keyword argument ``detailed_summary=True`` is passed to
+    build/run. A response of type **LuigiRunResult** has the following attributes:
+
+    Attributes:
+        * **summary_text_one_line** - One line summary of the progress.
+        * **summary_text** - Detailed summary of the progress.
+        * **status** - Integer Return Code. See \
+        `LuigiRetCodes <https://luigi.readthedocs.io/en/latest/api/luigi.execution_summary.html#luigi.execution_summary.LuigiRetCodes>`_ \
+        for what these codes mean.
+        * **worker** - Worker object.
+        * **execution_succeeded** - Boolean which is *True* if finally, there were no failed tasks.
+        * **scheduling_succeeded** - Boolean which is *True* if all the tasks were scheduled without errors.
+
     """
     def __init__(self, worker, worker_add_run_status=True):
         self.worker = worker
-        self.task_groups = _summary_dict(worker)
-        self.summary_detailed = summary(worker)
-        self.summary, smiley, _ = self._progress_summary()
-        self.status = True if smiley == ":)" else False
-        self.status_legacy = worker_add_run_status
-        self.response = self._response()
-
-    def _response(self):
-        if not execution_summary().legacy_run_result:
+        self.summary_text = summary(worker)
+        self.summary_text_one_line, self.status = self._progress_summary(worker)
+        self.scheduling_succeeded = worker_add_run_status
+        self.execution_succeeded = (True if self.status is LuigiRetCodes.SUCCESS or 
+                                            self.status is LuigiRetCodes.SUCCESS_WITH_RETRY
+                                    else False)
```

Comment:
Why `(True if cond else False)` when you can just `(cond)`?

Reply:
maybe even `self.status in (...)`?

Reply:
Ah yes! That would be much cleaner! On it! ⚡️